### DISCUSSION TITLE
Uncaught ReferenceError: React is not defined

### DISCUSSION THREAD

**Discussion by harshmangalam** [TanStack/query#4287] [General] [answered]
Getting error while adding solid-query in solidjs project

**Comment by TkDodo** [comment, marked as answer]
there have been issues with `vite-plugin-solid` that have since been fixed. Please make sure to use the latest version of both `solid-query` and `vite-plugin-solid`. See also:

- https://github.com/TanStack/query/pull/4211#issuecomment-1256915053
- https://github.com/solidjs/vite-plugin-solid/issues/57
